accelerators/corda/service-bus-integration/service-bus-listener/src/main/kotlin/net/corda/workbench/serviceBus/messaging/Connection.kt

```diff
@@ -15,46 +14,59 @@ import java.time.Duration
  */
 class Connection(private val registry: Registry) {
 
-    fun ingressQueueClient(): QueueClient {
+    fun ingressQueueClient(): ServiceBusSenderClient {
         val config = registry.retrieve(AzureConfig::class.java)
-        val connectionStringBuilder = ConnectionStringBuilder(config.endpoint, config.ingressQueue)
-        val retryPolicy = RetryExponential(Duration.ofSeconds(1),
-            Duration.ofSeconds(10),
-            5, "TRY5TIMES")
-        connectionStringBuilder.retryPolicy = retryPolicy
-        val sendClient = QueueClient(connectionStringBuilder, ReceiveMode.PEEKLOCK)
+        val sendClient = ServiceBusClientBuilder()
+            .connectionString(config.endpoint)
+            .retryOptions(AmqpRetryOptions()
+                .setMaxDelay(Duration.ofSeconds(10))
+                .setDelay(Duration.ofSeconds(1))
+                .setMaxRetries(5))
+            .sender()
+            .queueName(config.ingressQueue)
+            .buildClient()
 
         val ps = registry.retrieveOrElse(PrintStream::class.java, System.out)
         ps.println("Sender connected to queue ${config.ingressQueue} using endpoint ${config.endpoint}")
         return sendClient
     }
 
-    fun egressQueueClient(): QueueClient {
+    fun egressQueueClient(): ServiceBusSenderClient {
         val config = registry.retrieve(AzureConfig::class.java)
-        val sendClient = QueueClient(ConnectionStringBuilder(config.endpoint, config.egressQueue), ReceiveMode.PEEKLOCK)
+        val sendClient = ServiceBusClientBuilder()
+            .connectionString(config.endpoint)
+            .sender()
+            .queueName(config.egressQueue)
+            .buildClient()
 
         val ps = registry.retrieveOrElse(PrintStream::class.java, System.out)
         ps.println("Sender connected to queue ${config.egressQueue} using endpoint ${config.endpoint}")
         return sendClient
     }
 
-    fun ingressQueueReceiver(): IMessageReceiver {
+    fun ingressQueueReceiver(): ServiceBusReceiverAsyncClient {
```

Comment:
Looks like the only place this function is used is in https://github.com/Azure-Samples/blockchain-devkit/blob/master/accelerators/corda/service-bus-integration/service-bus-listener/src/main/kotlin/net/corda/workbench/serviceBus/messaging/IngressQueueListener.kt#L14 where the sync receive method is used. So, why return the async client here?